### PR TITLE
Fix #2250 by using a consistent naming strategy.

### LINKED ISSUE #2250: Building a catalogue with a mechanism with a different `SUFFIX` than filename crashes
Arbor 0.9, pip installed

**Describe the bug**
With filename `Cav2_1__0.mod`, and `SUFFIX  glia__glia_test_mods__Cav2_1__0`:

```
arbor-build-catalogue dbbs mod/
```

```
glia.exceptions.BuildCatalogueError: ABC errored out with exitcode 2

Command: arbor-build-catalogue glia_test_mods /tmp/tmpo_dcajwv
---- ABC output ----

Building debug code in '/tmp/tmp5wb69lrr'.
Build log:
[ 25%] modcc generating: /tmp/tmp5wb69lrr/build/generated/glia_test_mods/glia_test_mods_catalogue.cpp;/tmp/tmp5wb69lrr/build/generated/glia_test_mods/Cav2_1__0_cpu.cpp
Scanning dependencies of target glia_test_mods-catalogue
[ 25%] modcc generating: /tmp/tmp5wb69lrr/build/generated/glia_test_mods/glia_test_mods_catalogue.cpp;/tmp/tmp5wb69lrr/build/generated/glia_test_mods/Cav2_1__0_cpu.cpp
[ 50%] Building CXX object CMakeFiles/glia_test_mods-catalogue.dir/generated/glia_test_mods/glia_test_mods_catalogue.cpp.o
[ 75%] Building CXX object CMakeFiles/glia_test_mods-catalogue.dir/generated/glia_test_mods/Cav2_1__0_cpu.cpp.o


---- ABC error  ----

Warning: prefix '/opt/rh/devtoolset-10/root/usr/bin/c++' does not exist, falling back to 'c++'.
Traceback (most recent call last):
  File "/home/robin/.pyenv/versions/3.11.6/envs/arbor-dbbs/lib/python3.11/site-packages/arbor/bin/arbor-build-catalogue", line 318, in <module>
    sp.run(make_cmd, shell=True, check=True, capture_output = True, text = True)
  File "/home/robin/.pyenv/versions/3.11.6/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'make' returned non-zero exit status 2.
 Error:

c++: error: /tmp/tmp5wb69lrr/build/generated/glia_test_mods/Cav2_1__0_cpu.cpp: No such file or directory
c++: fatal error: no input files
compilation terminated.
make[2]: *** [CMakeFiles/glia_test_mods-catalogue.dir/build.make:83: CMakeFiles/glia_test_mods-catalogue.dir/generated/glia_test_mods/Cav2_1__0_cpu.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/glia_test_mods-catalogue.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

This is during a build managed by my CLI NMODl asset management tool, the log shows the `stdout` followed by `stderr`. These are the files created in the build dir's `build/generated/glia_test_mods`:

```
glia__glia_test_mods__Cav2_1__0.hpp  glia__glia_test_mods__Cav2_1__0_cpu.cpp  glia_test_mods_catalogue.cpp  glia_test_mods_catalogue.hpp
```

these filenames are derived from the `SUFFIX`, while the build log shows that it looks for them based on the original filename's derived location. It's fixed by lining up the `SUFFIX` statement with the filename.

### PR DESCRIPTION
**Pull request by thorstenhater**
Use the `module_name` string to name files consistently.

Fixes #2250

### CODE REVIEW

**Comment by thorstenhater** [comment]
@Helveg does this fix your troubles?

File: modcc/modcc.cpp [hunk 1]
```diff
@@ -188,99 +184,96 @@ int main(int argc, char **argv) {
...
+                        io::write_all(emit_gpu_cpp_source(m, popt), fn.string());
+                        fn.replace_extension(".cu");
+                        io::write_all(emit_gpu_cu_source(m, popt), fn.string());
                         break;
-                    case targetKind::cpu:
-                        io::write_all(emit_cpp_source(m, popt), outfile+"_cpu.cpp");
+                    }
+                    case targetKind::cpu: {
+                        fs::path fn = outdir / (mod + "_cpu");
+                        fn.replace_extension(".cpp");
```
Comment (boeschf) [inline modcc/modcc.cpp:250]:
```suggestion
                        fs::path fn = outdir / (mod + "_cpu.cpp");
```

**Review by boeschf — APPROVED** [review]
lgtm

File: modcc/modcc.cpp (hunk 1 above)
Comment (thorstenhater) [reply modcc/modcc.cpp:250]: I kind of like the symmetry. Yes, it's redundant, but I like it anyway ;)

**Pull request merged**